pkg/scheduler/plugins/multihierarchyelasticquota/quota_tree_wrapper.go

```diff
@@ -0,0 +1,476 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package multihierarchyelasticquota
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/koordinator-sh/koordinator/pkg/util"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v12 "k8s.io/apiserver/pkg/quota/v1"
+	"k8s.io/klog/v2"
+)
+
+type quotaNode struct {
+	quotaName         QuotaNameType
+	request           int64
+	sharedRatio       int64
+	minQuota          int64 //ensure的概念
+	runtimeQuota      int64
+	allowLentResource bool
+}
+
+func (node *quotaNode) String() string {
+	return fmt.Sprintf("qName:%s, req:%v, sharedRatio:%v, minQuota:%v, runtimeQuota:%v",
+		node.quotaName, node.request, node.sharedRatio, node.minQuota, node.runtimeQuota)
+}
+
+func NewQuotaNode(quotaName QuotaNameType, sharedRatio, request, min int64, allowLentResource bool) *quotaNode {
+	if sharedRatio == 0 {
+		return nil
+	}
+
+	return &quotaNode{
+		quotaName:         quotaName,
+		request:           request,
+		sharedRatio:       sharedRatio,
+		minQuota:          min,
+		runtimeQuota:      0,
+		allowLentResource: allowLentResource,
+	}
+}
+
+type quotaMapType map[QuotaNameType]v1.ResourceList
+
+//QuotaTreeWrapper 多资源维度quota runtime管理
+type QuotaTreeWrapper struct {
+	// When heightVersion is updated, this version++
+	heightVersion int64
+	// resource key vector
+	resourceKeys map[string]struct{}
+	// request limit for each group
+	groupReqLimit quotaMapType
+	// 去掉了quotaTree的包装，直接暴露quotaInfo，用作计算runtimeQuota
+	quotaInfoMap map[string]map[QuotaNameType]*quotaNode
+	// init equal mTotalResourceNoSys,then if req[id] > minQuota[id], Resource' -= minQuota[id]
+	totalResource v1.ResourceList //表示该层级的所有quota组可瓜分的资源，含(min)
+	lock          sync.RWMutex
+
+	treeName  string
+	quotaType string
+}
+
+func NewQuotaTreeWrapper(treeName string, quotaType string) *QuotaTreeWrapper {
+	return &QuotaTreeWrapper{
+		heightVersion: 1,
+		resourceKeys:  make(map[string]struct{}),
+		groupReqLimit: make(quotaMapType),
+		quotaInfoMap:  make(map[string]map[QuotaNameType]*quotaNode),
+		totalResource: util.NewResourceList(),
+		treeName:      treeName,
+		quotaType:     quotaType,
+	}
+}
+
+func (qtw *QuotaTreeWrapper) UpdateResourceKeys(resourceKeys map[string]struct{}) {
+	newResourceKey := make(map[string]struct{})
+	for resKey := range resourceKeys {
+		newResourceKey[resKey] = struct{}{}
+	}
+
+	qtw.lock.Lock()
+	defer qtw.lock.Unlock()
+
+	qtw.resourceKeys = newResourceKey
+	qtw.updateQuotaTreeDimensionByResourceKeys()
+}
+
+func (qtw *QuotaTreeWrapper) updateQuotaTreeDimensionByResourceKeys() {
```

Comment:
if do this way,  you will lose dimension which not in resourceKeys, especially in klog. It not our expectation. I was wonder why you delete quotaTree and quotaTreeWrapper's topo and put all logic in quotaTreeWrapper. It is a good choice to separate the core computing logic of runtime